src/topology/builder.rs

```diff
@@ -276,11 +276,8 @@ pub async fn build_pieces(
         debug!(component = %key, "Building new transform.");
 
         let mut schema_definitions = HashMap::new();
-        let merged_definition = if config.schema.enabled {
-            schema::merged_definition(&transform.inputs, config, &mut definition_cache)
-        } else {
-            schema::Definition::empty()
-        };
+        let merged_definition =
+            schema::merged_definition(&transform.inputs, config, &mut definition_cache);
```

Comment:
This seems like a problematic change? Unless further down the code, there are guards in place to make sure we don't enable schemas by default for everyone.

Reply:
The check for `config.schema.enabled` was moved to inside `merged_definition` since it's slightly more complicated logic now. (There are 4 possible schema's returned from a component, one for each combination of schema.enabled and log_namespace)

Reply:
To elaborate a bit more, when schema's are disabled, you still need to check which log namespaces an event can use (determined by the input schemas) to determine the definition, where before it was just `Definition::empty()`